lib/dry/core/deprecations.rb

```diff
@@ -33,17 +33,24 @@ class << self
         # Prints a warning
         #
         # @param [String] msg Warning string
-        def warn(msg, tag: nil)
-          tagged = "[#{tag || 'deprecated'}] #{msg.gsub(/^\s+/, '')}"
-          logger.warn(tagged)
+        # @param [String] tag Tag to help identify the source of the warning.
+        #   Defaults to "deprecated"
+        # @param [Integer] Caller frame to add to the message
+        def warn(msg, tag: nil, uplevel: nil)
+          caller_info = uplevel.nil? ? nil : caller[uplevel]
```

Comment:
this retrieves the whole stacktrace which is quite bad for performance. This is what I do in dry-monads to work around https://github.com/dry-rb/dry-monads/blob/18395296a304f1ac192232d2b87f532cf3facab6/lib/dry/monads/right_biased.rb#L258

Reply:
Thanks for pointing it out. I'll fix it very soon.

Reply:
No need, I already did :)